src/main/java/com/easypost/model/beta/Referral.java

```diff
@@ -0,0 +1,303 @@
+     *
+     * @param params Map of parameters.
+     * @return List<Referral> object.
+     * @throws EasyPostException when the request fails.
+     */
+    public static List<Referral> all(final Map<String, Object> params) throws EasyPostException {
+        return all(params, null);
+    }
+
+    /**
+     * List all Referral objects. This function requires the Partner User's API key.
+     *
+     * @param params Map of parameters.
+     * @param apiKey API key to use in request (overrides default API key).
+     * @return List<Referral> object.
+     * @throws EasyPostException when the request fails.
+     */
+    public static List<Referral> all(final Map<String, Object> params, String apiKey) throws EasyPostException {
+        Referral[] response = request(RequestMethod.GET, String.format("%s/%s", EasyPost.BETA_API_BASE,
+            "referral_customers"), params, Referral[].class, apiKey);
+
+        return Arrays.asList(response);
+    }
+
+    /**
+     * Add credit card to a referral user. This function requires the Referral User's API key.
+     *
+     * @param referralApiKey API key of the referral user.
+     * @param number Credit card number.
+     * @param expirationMonth Expiration month of the credit card.
+     * @param expirationYear Expiration year of the credit card.
+     * @param cvc CVC of the credit card.
+     *
+     * @return CreditCard object.
+     * @throws Exception when the request fails.
+     */
+    public static CreditCard addCreditCard(String referralApiKey, String number,
+        int expirationMonth, int expirationYear, String cvc) throws Exception {
+        return addCreditCard(referralApiKey, number, expirationMonth, expirationYear, cvc, Priority.PRIMARY);
+    }
+
+    /**
+     * Add credit card to a referral user. This function requires the Referral User's API key.
+     *
+     * @param referralApiKey API key of the referral user.
+     * @param number Credit card number.
+     * @param expirationMonth Expiration month of the credit card.
+     * @param expirationYear Expiration year of the credit card.
+     * @param cvc CVC of the credit card.
+     * @param priority Priority of this credit card.
+     *
+     * @return CreditCard object.
+     * @throws Exception when the request fails.
+     */
+    public static CreditCard addCreditCard(String referralApiKey, String number,
+        int expirationMonth, int expirationYear, String cvc, Priority priority) throws Exception {
+        String easypostStripeApiKey = retrieveEasypostStripeApiKey();
+        String stripeToken;
+
+        try {
+            stripeToken = createStripeToken(number, expirationMonth, expirationYear, cvc, easypostStripeApiKey);
+        } catch (Exception e) {
+            throw new Exception("Could not send card details to Stripe, please try again later", e);
+        }
+
+        return createEasypostCreditCard(referralApiKey, stripeToken, priority.toString());
```

Comment:
Let's go ahead and lowercase the `priority` here since the enum is all caps - just in case.